solana-programs/audius_eth_registry/src/processor.rs

```diff
@@ -27,14 +36,127 @@ impl Processor {
+            signature_data.recovery_id;
+
+        let message_data_offset = signature_offset + SecpSignatureOffsets::SECP_SIGNATURE_SIZE + 1;
+        instruction_data[message_data_offset..].copy_from_slice(&signature_data.message);
+
+        let num_signatures = 1;
+        instruction_data[0] = num_signatures;
+        let offsets = SecpSignatureOffsets {
+            signature_offset: signature_offset as u16,
+            signature_instruction_index: 0,
+            eth_address_offset: eth_address_offset as u16,
+            eth_address_instruction_index: 0,
+            message_data_offset: message_data_offset as u16,
+            message_data_size: signature_data.message.len() as u16,
+            message_instruction_index: 0,
+        };
+
+        let packed_offsets = offsets.try_to_vec();
+        instruction_data[1..data_start].copy_from_slice(&packed_offsets.unwrap());
+
+        return instruction_data;
+    }
+
+    /// Process [Convert i64 from Vec<u8>] ()
+    pub fn int_from_vec(
+        message: &Vec<u8>
+    ) -> i64 {
+        let mut intermediate_array = [0u8; 8];
+        intermediate_array[0..8].copy_from_slice(message);
+        return i64::from_le_bytes(intermediate_array);
+    }
+
+    /// Process [validate timestamp messages] ()
+    pub fn validate_timestamp_messages(
+        clock: &sysvar::clock::Clock,
+        message_1: &Vec<u8>,
+        message_2: &Vec<u8>,
+        message_3: &Vec<u8>,
+    ) -> Result<(), AudiusError> {
+        let timestamp_1: i64 = Self::int_from_vec(message_1);
+        let timestamp_2: i64 = Self::int_from_vec(message_2);
+        let timestamp_3: i64 = Self::int_from_vec(message_3);
+
+        if (clock.unix_timestamp - timestamp_1).abs() > MAX_TIME_DIFF_SECONDS
+            || (clock.unix_timestamp - timestamp_2).abs() > MAX_TIME_DIFF_SECONDS
+            || (clock.unix_timestamp - timestamp_3).abs() > MAX_TIME_DIFF_SECONDS
+        {
+            return Err(AudiusError::InvalidInstruction.into());
+        }
+
+        return std::result::Result::Ok(());
+    }
+
+
+    /// Process [Recover SECP Instructions]().
+    pub fn recover_secp_instructions(
+        instruction_info: &AccountInfo
+    ) -> Result<Vec<Instruction>, AudiusError> {
+        let mut v: Vec<Instruction> = Vec::new();
+        // Index of current instruction in tx
+        let index = sysvar::instructions::load_current_index(&instruction_info.data.borrow());
+        // Indicates no instructions present
+        if index == 0 {
+            return Err(AudiusError::Secp256InstructionLosing.into());
+        }
+        // Iterate over all instructions and recover SECP instruction
+        let mut iterator = 0;
+        while iterator < index {
+            let secp_instruction = sysvar::instructions::load_instruction_at(
+                iterator as usize,
+                &instruction_info.data.borrow(),
+            )
+            .unwrap();
```

Comment:
It is recommended to replace `unwrap` here with mapping error to a proper program error to avoid `panic`.